crates/ruff_linter/src/rules/pyupgrade/snapshots/ruff_linter__rules__pyupgrade__tests__UP050.py.snap

```diff
@@ -0,0 +1,322 @@
+---
+source: crates/ruff_linter/src/rules/pyupgrade/mod.rs
+---
+UP050.py:23:12: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+21 | ### Errors
+22 |
+23 | class C[T](Generic[_A]): ...
+   |            ^^^^^^^^^^^ UP050
+24 | class C[T](Generic[_B], str): ...
+25 | class C[T](int, Generic[_C]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+20 20 | 
+21 21 | ### Errors
+22 22 | 
+23    |-class C[T](Generic[_A]): ...
+   23 |+class C[T, _A]: ...
+24 24 | class C[T](Generic[_B], str): ...
+25 25 | class C[T](int, Generic[_C]): ...
+26 26 | class C[T](bytes, Generic[_D], bool): ...
+
+UP050.py:24:12: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+23 | class C[T](Generic[_A]): ...
+24 | class C[T](Generic[_B], str): ...
+   |            ^^^^^^^^^^^ UP050
+25 | class C[T](int, Generic[_C]): ...
+26 | class C[T](bytes, Generic[_D], bool): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+21 21 | ### Errors
+22 22 | 
+23 23 | class C[T](Generic[_A]): ...
+24    |-class C[T](Generic[_B], str): ...
+   24 |+class C[T, _B: int](str): ...
+25 25 | class C[T](int, Generic[_C]): ...
+26 26 | class C[T](bytes, Generic[_D], bool): ...
+27 27 | class C[T](Generic[_E], list[_E]): ...
+
+UP050.py:25:17: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+23 | class C[T](Generic[_A]): ...
+24 | class C[T](Generic[_B], str): ...
+25 | class C[T](int, Generic[_C]): ...
+   |                 ^^^^^^^^^^^ UP050
+26 | class C[T](bytes, Generic[_D], bool): ...
+27 | class C[T](Generic[_E], list[_E]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+22 22 | 
+23 23 | class C[T](Generic[_A]): ...
+24 24 | class C[T](Generic[_B], str): ...
+25    |-class C[T](int, Generic[_C]): ...
+   25 |+class C[T, _C: (str, bytes)](int): ...
+26 26 | class C[T](bytes, Generic[_D], bool): ...
+27 27 | class C[T](Generic[_E], list[_E]): ...
+28 28 | class C[T](list[_F], Generic[_F]): ...
+
+UP050.py:26:19: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+24 | class C[T](Generic[_B], str): ...
+25 | class C[T](int, Generic[_C]): ...
+26 | class C[T](bytes, Generic[_D], bool): ...
+   |                   ^^^^^^^^^^^ UP050
+27 | class C[T](Generic[_E], list[_E]): ...
+28 | class C[T](list[_F], Generic[_F]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+23 23 | class C[T](Generic[_A]): ...
+24 24 | class C[T](Generic[_B], str): ...
+25 25 | class C[T](int, Generic[_C]): ...
+26    |-class C[T](bytes, Generic[_D], bool): ...
+   26 |+class C[T, _D](bytes, bool): ...
```

Comment:
this is an incorrect fix because the `_D` `TypeVar` has a `default` argument. The proper fix here would be

```suggestion
26    |-class C[T](bytes, Generic[_D], bool): ...
   26 |+class C[T, _D = int](bytes, bool): ...
```

We don't currently handle `default` keywords at all in our PEP-695 autofixes; it's one of the outstanding items in https://github.com/astral-sh/ruff/issues/15642. For now, for this rule, I think we should just make sure that we don't try to autofix any classes that are generic over any `TypeVar`s with the `default` argument.